cloud/passportauth/src/auth/PassportAuth.ts

```diff
@@ -103,13 +106,29 @@ export class PassportAuth implements Auth {
   }
 
   /**
+   * Checks if the current user has the role required to access a resource
+   *
+   * @param user - user data
+   * @param role - The required role to access a resource
+   * @returns Returns true/false if the user is authorized to access a resource
+   */
+  public hasResourceRole(user: any, role?: string) {
```

Comment:
[Idea/Optional] Can this also be pulled out to actual service. Users will be free to implement their own role matching algorithm